internal/gps/source_manager.go

```diff
@@ -578,9 +566,29 @@ func (sm *SourceMgr) InferConstraint(s string, pi ProjectIdentifier) (Constraint
 		return version.Unpair(), nil
 	}
 
+	// Revision, possibly abbreviated
```

Comment:
Originally the function checked if the string given was a revision first (note the comment at the top of the file). Unless we need to change that behavior(?), let's move this back to the top where we are replacing the old code.

Reply:
Okay then let's just update the comment to reflect the new order.